fossology/fossology
# feat(export): Export Copyrights

## Description

Added new functionality for exporting copyrights to License List. Thus renamed the page to more general Export List.

The page can now export License List (as before) and Copyright List as well. The Copyright List has a filter to export all copyrights or to export copyrights from files which do not have license.

Files without license here refers to:
1. File had no license finding by agents and no license was added by users.
2. File had license findings by agents but were either removed or file was marked as irrelevant.

### Changes

1. Rename page License List to Export List and file `src/www/ui/ui-license-list.php` to `src/www/ui/ui-export-list.php`.
1. Rename template `src/www/ui/template/ui-license-list-form.html.twig` to `src/www/ui/template/ui-export-list.html.twig`.
1. Add copyright exports to `UIExportList`.
1. Fix the links in UI.

## How to test

1. Upload a package without running any agents and check if "Export List" can handle it.
1. Run only copyright agent on the package and do some manual clearing.
    1. All the copyrights should be exported for both options.
1. Run some scanners and do some clearing.
    1. The copyright list should reflect the changes when exporting files without licenses.

## Todo
Need new issues for following:
- [x] Add copyright export to `fo_nomos_license_list`. (#1715)
- [x] Add copyright export to REST API. (#1716)

Closes #1687
Closes #1539
